IntegrationTests/MeasurementSchedulerIntegrationTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using DatabaseAdapters;
+using DatabaseAdapters.Repositories;
+using DatabaseAdapters.Repositories.Test;
+using Domain.Entities;
+using Domain.Repositories;
+using Domain.Services;
+using Domain.Services.External;
+using Domain.Services.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Moq;
+using TemperatureSystem.HostedServices;
+using Xunit;
+
+namespace IntegrationTests;
+
+public class MeasurementSchedulerIntegrationTests : IDisposable
+{
+    private readonly ServiceProvider _serviceProvider;
```

Comment:
When possible use primary constructors instead of private fields and standard constructors.